src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1067,7 +1068,10 @@ def _set_language_and_task(language, task, is_multilingual, generation_config):
                     "to `generate`. Either set the language using the `forced_decoder_ids` in the model config, "
                     "or update the generation config as per the instructions https://github.com/huggingface/transformers/issues/25084#issuecomment-1664398224"
                 )
-            language = language.lower()
+            if isinstance(language, str):
```

Comment:
I would be in favour of moving this normalisation of the `language` argument to the `_retrieve_init_tokens` method. It can be executed just before we map the `language` string argument to the langauge id token: https://github.com/huggingface/transformers/blob/9b0a8ea7d1d6226b76cfdc645ce65e21157e2b50/src/transformers/models/whisper/generation_whisper.py#L1138

Reply:
@sanchit-gandhi But this is already the `_retrieve_init_tokens` function. But yes, I'll move it here to normalize the list of languages once it's already been ensured that it's a list.